org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -3322,10 +3322,28 @@ public final class JavaCore extends Plugin {
 	 * @category OptionValue
 	 */
 	public static final String VERSION_CLDC_1_1 = "cldc1.1"; //$NON-NLS-1$
-	private static List<String> allVersions = Collections.unmodifiableList(Arrays.asList(VERSION_CLDC_1_1, VERSION_1_1, VERSION_1_2, VERSION_1_3, VERSION_1_4, VERSION_1_5,
+	private static final List<String> allVersions = Collections.unmodifiableList(Arrays.asList(VERSION_CLDC_1_1, VERSION_1_1, VERSION_1_2, VERSION_1_3, VERSION_1_4, VERSION_1_5,
```

Comment:
The modern version is  `List.of(...)`
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/List.html#of(E...)

Reply:
True. But it shall be noted that the returned List implements an extremely rigor null-contract. It'll throw an NPE when asked whether it `contains(null)` instead of just returning false. I learned the hard way that this *can* be quite  painful.

Reply:
Thanks Sebastian, will leave "as is".